pymongo/pool.py

```diff
@@ -1549,7 +1586,8 @@ def connect(self, handler: Optional[_MongoClientErrorHandler] = None) -> Connect
                 )
 
             if isinstance(error, (IOError, OSError, SSLError)):
-                _raise_connection_failure(self.address, error)
+                details = _get_timeout_details(self.opts)
+                _raise_connection_failure(self.address, error, timeout_details=details)
```

Comment:
I see _configured_socket() also calls _raise_connection_failure(). Will this then add timeout_details to the error twice?

Reply:
Good catch. I'll add a check in `_raise_connection_failure` to ensure it doesn't add `timeout_details` if they're already in the message.